packages/strapi-plugin-upload/admin/src/components/BrowseAssets/index.js

```diff
@@ -53,6 +54,12 @@ const BrowseAssets = () => {
   const limit = parseInt(params._limit, 10) || 10;
   const start = parseInt(params._start, 10) || 0;
 
+  const handleListCardClick = id => {
+    handleFileSelection({
+      target: { name: parseInt(id, 10), value: selectedFiles.findIndex(file => file.id === id) },
```

Comment:
When you use mongoose your ids are `String` so instead you should do a `toString()` in your `findIndex` otherwise it won't work with the other orm.